### PR TITLE
Externalise option to store more in results_root

### PR DESCRIPTION
**Pull request by sxa**
Needed to retain the full set of results when the JCK is executed via STF. Ref: https://github.com/AdoptOpenJDK/stf/issues/22

### CODE REVIEW

File: stf.core/scripts/stf.pl [hunk 1]
```diff
@@ -162,20 +162,41 @@
...
+my $delete_results_root = $FALSE;
 
 my @results_dirs;
 
+if ( length $retain_limit ) {
+   $results_retention_limit = $retain_limit;
+}
+
+if ( length $retain_number ) {
+   if ( $retain_number == 0 ) {
```
Comment (lumpfish) [inline stf.core/scripts/stf.pl]: Personally I'd prefer not having the magic number 0 meaning 1000000.  Apart from being cryptic it means you can't specify 0 to mean 'keep no results'.

**Comment by lumpfish** [comment]
Please can you also update the help output in Stf.java for the new options.

File: stf.core/scripts/stf.pl (hunk 1 above)
Comment (sxa) [reply stf.core/scripts/stf.pl]: Fair enough - I did also consider using -1 but happy to just remove the clause altogether

**Comment by sxa** [comment]
I'm reluctant to add it to the help output as a user will never need to use that option to Stf.java - it's only used by the perl script but the java code needs to tolerate it's presence in the properties file that the perl script requires for a default value

**Comment by sxa** [comment]
OK I've added some (very brief) help into the `stf.pl -help` output which seems the best place to put it at the moment.

**Comment by lumpfish** [comment]
Looks good

**Pull request merged**